### ISSUE TITLE
Vault UI - top level secrets not displaying

### ISSUE THREAD

**Issue by censored** [hashicorp/vault#3348]
**Environment:**
* Vault Version: Vault v0.8.2+pro ('f6153a5487af04f3e05f4f2f4f00ea1f4e564412+CHANGES')
* Operating System/Architecture: Ubuntu Xenial 16.04.1 LST x86_64

**Vault Config File:**
N/A
**Startup Log Output:**
N/A

**Expected Behavior:**
Write a top level key under a secret path via the CLI and be able to view it in the UI.

**Actual Behavior:**
The UI throws an error 
```
TypeError: Cannot read property 'secret' of undefined
    at le (vendor-1406a607d03c7cd246c522d47589b1ed.js:2138)
    at e.calculateCacheKey (vendor-1406a607d03c7cd246c522d47589b1ed.js:2700)
    at o._hydrateUnsuppliedQueryParams (vendor-1406a607d03c7cd246c522d47589b1ed.js:2668)
    at o._prepareQueryParams (vendor-1406a607d03c7cd246c522d47589b1ed.js:2660)
    at o._doTransition (vendor-1406a607d03c7cd246c522d47589b1ed.js:2655)
    at o.transitionTo (vendor-1406a607d03c7cd246c522d47589b1ed.js:2636)
    at o.click (vault-enterprise-f96dd26c26c8a7ec4b710adf835326e0.js:99)
    at o.trigger (vendor-1406a607d03c7cd246c522d47589b1ed.js:3148)
    at o.r [as trigger] (vendor-1406a607d03c7cd246c522d47589b1ed.js:3048)
    at e.run (vendor-1406a607d03c7cd246c522d47589b1ed.js:1392)
```

**Steps to Reproduce:**
1. Generate a new secret without a path under the secret backend:
```
steph@purgatory:~$ VAULT_ADDR=https://vault.dev.xx.com vault write cubbyhole test=true
Success! Data written to: cubbyhole
steph@purgatory:~$ VAULT_ADDR=https://vault.dev.xx.com vault read cubbyhole 
Key 	Value
--- 	-----
test	true
```
Attempt to view the key via the UI and you can see it is blank, and clicking on it generates an error

https://drive.google.com/open?id=0ByLHE0wpFO0MZklzSDUybTQ0bHc
https://drive.google.com/open?id=0ByLHE0wpFO0Md3kwcWtBWENJeW8

**Important Factoids:**
I expect most of the time people won't be doing this, but knowing the user base I will be exposing this to, it is quite possible it may happen. 

Viewing and removing this secret via the CLI works fine:

``` 
steph@purgatory:~$ VAULT_ADDR=https://vault.dev.xx.com vault write cubbyhole test=true
Success! Data written to: cubbyhole
steph@purgatory:~$ VAULT_ADDR=https://vault.dev.xx.com vault read cubbyhole 
Key 	Value
--- 	-----
test	true

steph@purgatory:~$ VAULT_SKIP_VERIFY=true VAULT_ADDR=https://vault.dev.xx.com vault delete cubbyhole
Success! Deleted 'cubbyhole' if it existed.
steph@purgatory:~$ VAULT_ADDR=https://vault.dev.xx.com vault read cubbyhole 
No value found at cubbyhole
```
Chrome Versions
Version 56.0.2924.87 (64-bit)
Version 60.0.3112.113 (Official Build) (64-bit)
Firefox Version 49.0

**References:**

**Comment by meirish** [comment]
HI @censored - I got a fix in for this and it should go out with the next release. We're not currently supporting creating these secrets in the UI (though we'll likely add that in the near term), but I've changed things so that you can view / edit / delete them, opting to use the label `(self)` for now. I've attached a screenshot of what that looks like:
![screen shot 2017-09-19 at 9 25 51 am](https://user-images.githubusercontent.com/39469/30726666-de3668ca-9f11-11e7-8ac4-7fd46ff14178.png)

**Comment by meirish** [comment]
This landed in 0.8.3
